File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/views/SegmentedControlDemoVG.kt
package com.lightningkite.kwifttemplate.shared.views

import android.view.View
import com.lightningkite.kwift.observables.actual.bind
import com.lightningkite.kwift.observables.shared.StandardObservableProperty
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwift.views.shared.ViewGenerator
import com.lightningkite.kwifttemplate.xml.SegmentedControlDemoXml

class SegmentedControlDemoVG() : ViewGenerator() {
    override val title: String get() = "Segmented Control Demo"

    override fun generate(dependency: ViewDependency): View {
        val xml = SegmentedControlDemoXml()
        val view = xml.setup(dependency)

        xml.tabs.bind(listOf("A", "B", "C"), StandardObservableProperty(0))
        xml.tabs2.bind(listOf("A", "B", "C", "D", "E", "F", "G"), StandardObservableProperty(0))

        return view
    }
}

File: ios/Kwift Template/com/lightningkite/kwifttemplate/shared/views/SegmentedControlDemoVG.swift
//Package: com.lightningkite.kwifttemplate.shared.views
//Converted using Kwift2

import Foundation
import Kwift
import KwiftMaps



public class SegmentedControlDemoVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Segmented Control Demo"
        }
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = SegmentedControlDemoXml()
        var view = xml.setup(dependency)
        xml.tabs.bind(["A", "B", "C"], StandardObservableProperty(0))
        xml.tabs2.bind(["A", "B", "C", "D", "E", "F", "G"], StandardObservableProperty(0))
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        super.init()
    }
}
 
